exec/cnex/lib/posix.c

```diff
@@ -0,0 +1,781 @@
+
+void posix_fork(struct tagTExecutor *exec)
+{
+    push(exec->stack, wrap(fork()));
+}
+
+void posix_fpathconf(struct tagTExecutor *exec)
+{
+    Number name = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fpathconf(number_to_sint32(fildes), number_to_sint32(name))));
+}
+
+void posix_fsync(struct tagTExecutor *exec)
+{
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(fsync(number_to_sint32(fildes))));
+}
+
+void posix_ftruncate(struct tagTExecutor *exec)
+{
+    Number length = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(ftruncate(number_to_sint32(fildes), number_to_uint64(length))));
+}
+
+void posix_getegid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getegid())));
+}
+
+void posix_geteuid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(geteuid())));
+}
+
+void posix_getgid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getgid())));
+}
+
+void posix_gethostname(struct tagTExecutor *exec)
+{
+    char name[256];
+    Cell *r = wrap(gethostname(name, 256));
+
+    push(exec->stack, cell_fromCString(name));
+    push(exec->stack, r);
+}
+
+void posix_getlogin(struct tagTExecutor *exec)
+{
+    char *r = getlogin();
+    push(exec->stack, cell_fromCString(r));
+}
+
+void posix_getpgid(struct tagTExecutor *exec)
+{
+    Number pid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(getpgid(number_to_sint32(pid))));
+}
+
+void posix_getpgrp(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getpgrp())));
+}
+
+void posix_getpid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getpid())));
+}
+
+void posix_getppid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getppid())));
+}
+
+void posix_getpriority(struct tagTExecutor *exec)
+{
+    Number who = top(exec->stack)->number; pop(exec->stack);
+    Number which = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(getpriority(number_to_sint32(which), number_to_sint32(who))));
+}
+
+void posix_getsid(struct tagTExecutor *exec)
+{
+    Number pid = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(getsid(number_to_sint32(pid))));
+}
+
+void posix_getuid(struct tagTExecutor *exec)
+{
+    push(exec->stack, cell_fromNumber(number_from_sint32(getuid())));
+}
+
+void posix_isatty(struct tagTExecutor *exec)
+{
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, cell_fromBoolean(isatty(number_to_sint32(fildes)) != 0));
+}
+
+void posix_kill(struct tagTExecutor *exec)
+{
+    Number pid = top(exec->stack)->number; pop(exec->stack);
```

Comment:
Are you sure these parameters are popped in the correct order?

Reply:
No.  After looking at posix.cpp, I do believe I had these in the wrong order.  Fixed.